admin/server/projects.go

```diff
@@ -680,42 +681,62 @@ func (s *Server) GetProjectVariables(ctx context.Context, req *adminv1.GetProjec
 		return nil, status.Error(codes.PermissionDenied, "does not have permission to read project variables")
 	}
 
-	return &adminv1.GetProjectVariablesResponse{Variables: proj.ProdVariables}, nil
+	var environment *string
+	if !req.ForAllEnvironments {
+		environment = &req.Environment
+	}
+	vars, err := s.admin.DB.FindProjectVariablesByEnvironment(ctx, proj.ID, environment)
```

Comment:
I think the expected behavior here is that it resolves the variables for the provided environment (i.e. also incorporating variables that are not environment-specific), not that it returns only the variables configured for that environment.

Resolving enables use of this endpoint to fetch variables for a specific environment, e.g. for getting all variables for the "dev" environment on local.